enhancements/running-on-the-host.md

```diff
@@ -0,0 +1,167 @@
+
+## Motivation
+
+Running the nodes completely out of VPN does not work with the current design. In order to configure the router, we need to fetch the configuration from the API server, but in order to reach the API server we need to configure the router.
+
+To solve this chicken-egg problem we need an operating mode where the router is allowed to start as soon as possible and to be fed with a static configuration used to provide the basic connectivity required to bootstrap kubernetes.
+
+## Goals
+
+- Running the router before the kubelet and the cluster's primary CNI
+- Providing a way to configure the router statically
+- Providing a way to extend the static configuration with a configuration provided via the current kubernetes API
+
+## Non Goals
+
+- Having an alternative way to provide the static configuration that involves distritbuting the configuration to the routers
+
+## Proposal
+
+### User stories
+
+- As a cluster admin, I want to deploy the router to the nodes of the cluster with enough configuration to allow it to provide the basic connectivity.
+
+- As a cluster admin, I want to use Kubernetes CRDs to provision additional networks to reach / expose different classes of services or implement multitenancy.
+
+### Design Details
+
+The general idea is to run the same set of containers we have in pods today, but on the host. This includes:
+
+- running the router pod, in charge of providing the network control plane with FRR and the data plane with veths and the underlay interface
+- running the controller pod, in charge of reading the static configuration and receiving the additional configuration when the kubernetes api is available
+
+#### Handling the lifecycle of the components
+
+The router and the controller need to run as systemd units. This allows us to run them before the kubelet and the CNI start.
+Once the machinery is implemented, alternative ways to run them can be added for those environments where systemd is not enabled.
+
+#### Running on the host
+
+Podman provides a way to maintain the current pod structure while running the pods as systemd units. This allow us to keep reasoning of pods and
+to have a good parallel between the "Kubernetes mode" and the "Host mode".
+
+Because of this, the initial implementation will be based on Podman. As per systemd above, when we have a way of running the pods on the host, it should be
+relatively easy to adapt to other container runtimes.
+
+#### Interactions
+
+There are some interactions between the controller process and the router (and the surrounding system) that need to be adapted to the new scenario:
+
+##### Retrieving the router's network namespace
+
+We are using the container runtime api to retrieve the network namespace of the pod.
+When running on the host, the target namespace can be reitrieved by telling podman to fill a pid file, and by using the pid to retrieve the network namespace of the process.
+
```

Comment:
Is it possible to run the controller as the router pod so they share the network namespace ? I understand that is not possible becuase the config reload that restart the router pod, but maybe with alternative reload mechanism they can live together so network namespace discovery is not needed.

Reply:
sorry but I didn't get the ask. The controller runs as a different pod as it does on k8s, because it must be hostnetworked in order to move interfaces around and create veths and move one leg of the veth in the network ns of the router

Reply:
Ahh wait right... you can disregard this, we don't have host network at routing pod since everything runs at the router namespace.